ghcide/test/exe/Main.hs

```diff
@@ -538,12 +538,17 @@ diagnosticTests = testGroup "diagnostics"
             , "foo = 1 {-|-}"
             ]
       _ <- createDoc "Foo.hs" "haskell" fooContent
+#if MIN_VERSION_ghc(9,0,1)
+      -- Haddock parse errors are ignored on ghc-9.0.1
```

Comment:
This means that we no longer need to parse twice (todo in a separate PR).

Reply:
Are the hlint related problems mentioned here resolved as well? https://github.com/haskell/haskell-language-server/blob/e0a464282db913b6c7e0f2ec4504d24e2bcfb4b1/ghcide/src/Development/IDE/Core/Rules.hs#L198-L206

Reply:
Ah, good question! I think it might, but I'm not sure. We can test it separately